cmd/minikube/cmd/start_flags.go

```diff
@@ -191,7 +192,8 @@ func initDriverFlags() {
 	startCmd.Flags().Bool("vm", false, "Filter to use only VM Drivers")
 
 	// kvm2
-	startCmd.Flags().String(kvmNetwork, "default", "The KVM network name. (kvm2 driver only)")
+	startCmd.Flags().String(kvmNetwork, "default", "The KVM default network name. (kvm2 driver only)")
+	startCmd.Flags().String(kvmPrivateNetwork, "", "The KVM private network name. (kvm2 driver only) (default: 'mk-<cluster_name>')")
```

Comment:
what is the difference of kvmNetwork and kvmPrivateNetwork
do u mind clearifying it in the help text ?


can a user specify both of them ?

how about adding some docs about it in the kvm page ?
https://minikube.sigs.k8s.io/docs/drivers/kvm2/

Reply:
***kvmNetwork*** was there from earlier, i've just added a more descriptive message to clearly distinguish it from the new kvmPrivateNetwork flag;
*kvmNetwork* references the ***default*** kvm network - the one that comes with standard libvirt installation to support nat, and is always called "default"; i don't think one can change this name for libvirt (eg, via kvmNetwork flag) and in the code, we just check if that network exists and if is active - ie, we do not try to create 'default' network and fail if it does not already exist!

i only guess that it was maybe added with the idea if it could be customised in libvirt sometime in the future (if anyone would want that)? *if it is not used - better removed*

on the other hand, the new kvmPrivateNetwork allows user to specify a name for the custom dedicated kvm cluster network that we make possible with this pr

proposal for code cleanup and docs update (maybe in a separate issue, not to block this one?):
- remove (unused) kvmNetwork flag and all references (incl the one in the cluster config) used also to pass it to the kvm driver, meant to "[re]defines" 'default' kvm network
- see if we need/use the 'default' kvm network at all - currently, each kvm vm/cluster is created with ***two*** interfaces: the 'default' one and the other is fixed 'minikube-net' (while after this pr each kvm cluster would have its own custom private kvm network):
https://github.com/kubernetes/minikube/blob/63a8f285bffa71364da2358b661f9ff796e90f1d/pkg/drivers/kvm/domain.go#L69-L78
- update the docs accordingly

i'm happy to check if we need/use the 'default' kvm network at all and do the above-proposed actions

Reply:
@prezha I suggest we use the existing
```
      --network='': network to run minikube with. Only available with the docker/podman drivers. If left empty, minikube will create a new network.
```
flag except we update the Help text to say "now it is used by docker/podman and KVM"

Reply:
@medya yes, that makes sense: i've change pr so that it now uses the existing '--network' flag for private kvm network